### PR TITLE
Make call-preview endpoint work without working faucet or with empty …

### PR DESCRIPTION
**Pull request by siy**
[Fix `call-preview` endpoint.](https://radixdlt.atlassian.net/browse/PRO-3)

Existing `call-preview` endpoint is unable to provide a preview if the faucet is not available or empty (i.e. in production environment). This PR addresses this issue and makes the endpoint work regardless of the presence of the faucet.

### CODE REVIEW

File: core/src/test/java/com/radixdlt/rev2/nofaucet/CallPreviewWithoutFaucetTest.java [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+                            ImmutableList.of(
+                                new GenesisStakeAllocation(
+                                    UInt32.fromNonNegativeInt(0), Decimal.ONE)))))),
+            Decimal.ofNonNegative(0L),
+            ImmutableList.of());
+
+    return DeterministicTest.builder()
+        .addPhysicalNodes(PhysicalNodeConfig.createBatch(1, true))
+        .messageSelector(firstSelector())
+        .messageMutator(MessageMutator.dropTimeouts())
```
Comment (dhedey) [inline core/src/test/java/com/radixdlt/rev2/nofaucet/CallPreviewWithoutFaucetTest.java:130]: Why are we dropping timeouts?

File: core/src/test/java/com/radixdlt/rev2/nofaucet/CallPreviewWithoutFaucetTest.java [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+import com.radixdlt.rev2.Decimal;
+import com.radixdlt.rev2.ScryptoConstants;
+import com.radixdlt.rev2.protocol.ProtocolUpdateTestUtils;
+import com.radixdlt.utils.UInt32;
+import com.radixdlt.utils.UInt64;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+public final class CallPreviewWithoutFaucetTest {
```
Comment (dhedey) [inline core/src/test/java/com/radixdlt/rev2/nofaucet/CallPreviewWithoutFaucetTest.java:95]: Out of interest, why didn't we use a test under `/api/core`, e.g. creating a `TransactionCallPreviewTest.java` similiar to `TransactionPreviewTest.java`?

In fact, can I request that we re-write the test there?

Comment (dhedey) [reply core/src/test/java/com/radixdlt/rev2/nofaucet/CallPreviewWithoutFaucetTest.java:95]: You can then configure the test to run different scenarios, as per `test_core_api_can_submit_and_commit_transaction_after_running_all_scenarios`

File: core/src/test/java/com/radixdlt/rev2/nofaucet/CallPreviewWithoutFaucetTest.java [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+  @Rule public TemporaryFolder folder = new TemporaryFolder();
+
+  private DeterministicTest createTest(Module... extraModules) {
```
Comment (dhedey) [inline core/src/test/java/com/radixdlt/rev2/nofaucet/CallPreviewWithoutFaucetTest.java:98]: Lots of this configuration seems irrelevant and prone to breaking - can we either use some helper method like
`GenesisBuilder.createTestGenesisWithNumValidators` ?

File: core/src/test/java/com/radixdlt/rev2/nofaucet/CallPreviewWithoutFaucetTest.java [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+    final var accountLockerCall =
+        new TransactionCallPreviewRequest()
+            .network(Network.INTEGRATIONTESTNET.getLogicalName())
+            .target(
+                new BlueprintFunctionTargetIdentifier()
+                    .packageAddress(addressing.encode(ScryptoConstants.RESOURCE_PACKAGE_ADDRESS))
+                    .blueprintName("FungibleResourceManager")
+                    .functionName("create")
+                    .type(TargetIdentifierType.FUNCTION))
+            .addArgumentsItem("4d220000")
```
Comment (dhedey) [inline core/src/test/java/com/radixdlt/rev2/nofaucet/CallPreviewWithoutFaucetTest.java:165]: (Side note, irrelevant to this PR!) Hah oh - I forgot how awful this interface is. I believe there's a ticket in the backlog to allow using manifest syntax for this in a new endpoint...

File: core/src/test/java/com/radixdlt/rev2/nofaucet/CallPreviewWithoutFaucetTest.java [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+            .addArgumentsItem("4d0100")
+            .addArgumentsItem("4d0700")
+            .addArgumentsItem("4d2106220000220000220000220000220000220000")
+            .addArgumentsItem("4d2102230c2100230c2200")
+            .addArgumentsItem("4d220000");
+
+    final var coreApiHelper = new ProtocolUpdateTestUtils.CoreApiHelper();
```
Comment (dhedey) [inline core/src/test/java/com/radixdlt/rev2/nofaucet/CallPreviewWithoutFaucetTest.java:172]: We're not in a protocol update test, so feels a little weird to use this.

File: core/src/test/java/com/radixdlt/rev2/nofaucet/CallPreviewWithoutFaucetTest.java [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+    try (var test = createTest(coreApiHelper.module())) {
+      // Arrange: Start a single node network
+      test.startAllNodes();
+
+      // Act: Preview a transaction
+      final var callBeforeBottlenose =
```
Comment (dhedey) [inline core/src/test/java/com/radixdlt/rev2/nofaucet/CallPreviewWithoutFaucetTest.java:178]: How do we know we're actually before bottlenose?

Comment (dhedey) [reply core/src/test/java/com/radixdlt/rev2/nofaucet/CallPreviewWithoutFaucetTest.java:178]: I'd much rather we make it explicit / clear.

e.g. we can set a really long epoch in genesis so that anemone doesn't get enacted (for now) and add a comment explaining _why_ we have set the long epoch.

(And then, in the long-term, we should be able to generally reconfigure protocol update running in the test-runner, e.g. create a: `integration_test_no_scenarios_protocol_config() -> ProtocolConfig`, or use the `ProtocolUpdateContentOverrides` mechanism - or something, like we can now do on the rust side.

File: core/src/test/java/com/radixdlt/rev2/nofaucet/CallPreviewWithoutFaucetTest.java (hunk 3 above)
Comment (dhedey) [reply core/src/test/java/com/radixdlt/rev2/nofaucet/CallPreviewWithoutFaucetTest.java:98]: In fact, if we just use a core api test base, we can get rid of lots of this.

**Review by dhedey — CHANGES_REQUESTED** [review]
Nice functional changes - but I'd like us to make the test less boiler-platey and make the hacks clearer with some comments 👍

File: core/src/test/java/com/radixdlt/rev2/nofaucet/CallPreviewWithoutFaucetTest.java (hunk 1 above)
Comment (siy) [reply core/src/test/java/com/radixdlt/rev2/nofaucet/CallPreviewWithoutFaucetTest.java:130]: These are leftovers from the test I've used as a base.

File: core/src/test/java/com/radixdlt/rev2/nofaucet/CallPreviewWithoutFaucetTest.java (hunk 2 above)
Comment (siy) [reply core/src/test/java/com/radixdlt/rev2/nofaucet/CallPreviewWithoutFaucetTest.java:95]: I'll move test there, but this is rather organizational change. This test is specific in the sense that it has different genesis, hence reusing existing classes gets quite inconvenient.

File: core/src/test/java/com/radixdlt/rev2/nofaucet/CallPreviewWithoutFaucetTest.java (hunk 3 above)
Comment (siy) [reply core/src/test/java/com/radixdlt/rev2/nofaucet/CallPreviewWithoutFaucetTest.java:98]: Yes, but this test requires the faucet to be empty, which is not the case for the majority of tests.

File: core/src/test/java/com/radixdlt/rev2/nofaucet/CallPreviewWithoutFaucetTest.java (hunk 6 above)
Comment (siy) [reply core/src/test/java/com/radixdlt/rev2/nofaucet/CallPreviewWithoutFaucetTest.java:178]: Actually, this is also leftover from the test I've used as a base.

File: core/src/test/java/com/radixdlt/rev2/nofaucet/CallPreviewWithoutFaucetTest.java (hunk 2 above)
Comment (dhedey) [reply core/src/test/java/com/radixdlt/rev2/nofaucet/CallPreviewWithoutFaucetTest.java]: > I'll move test there, but this is rather organizational change. This test is specific in the sense that it has different genesis, hence reusing existing classes gets quite inconvenient.

`test_core_api_can_submit_and_commit_transaction_after_running_all_scenarios` which I mentioned above uses a different genesis alongside various helpers and builders which reduces the boilerplate for such changes. It can be used as an example for such a change.

File: core/src/test/java/com/radixdlt/api/core/CallPreviewTest.java [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+import com.radixdlt.api.core.generated.models.TransactionStatus;
+import com.radixdlt.crypto.ECKeyPair;
+import com.radixdlt.genesis.*;
+import com.radixdlt.modules.StateComputerConfig;
+import com.radixdlt.networks.Network;
+import com.radixdlt.rev2.Decimal;
+import com.radixdlt.rev2.ScryptoConstants;
+import org.junit.Test;
+
+public final class CallPreviewTest extends DeterministicCoreApiTestBase {
```
Comment (dhedey) [inline core/src/test/java/com/radixdlt/api/core/CallPreviewTest.java:85]: Much better without all the boilerplate, thanks!

**Review by dhedey — APPROVED** [review]
